snapshot/policy/scheduling_policy.go

```diff
@@ -72,6 +77,49 @@ func (p *SchedulingPolicy) Merge(src SchedulingPolicy) {
 
 	p.TimesOfDay = SortAndDedupeTimesOfDay(
 		append(append([]TimeOfDay(nil), src.TimesOfDay...), p.TimesOfDay...))
+
+	if !p.Manual {
+		p.Manual = src.Manual
+	}
+}
+
+// IsManualSnapshot returns the SchedulingPolicy manual value from the given policy tree.
+func IsManualSnapshot(policyTree *Tree) bool {
+	return policyTree.EffectivePolicy().SchedulingPolicy.Manual
+}
+
+// SetManual sets the manual setting in the SchedulingPolicy on the given source.
+func SetManual(ctx context.Context, rep repo.RepositoryWriter, sourceInfo snapshot.SourceInfo) error {
+	// Get existing defined policy for the source
+	p, err := GetDefinedPolicy(ctx, rep, sourceInfo)
+
+	switch {
+	case errors.Is(err, ErrPolicyNotFound):
+		p = &Policy{}
+	case err != nil:
+		return errors.Wrap(err, "could not get defined policy for source")
+	}
+
+	p.SchedulingPolicy.Manual = true
+
+	if err := SetPolicy(ctx, rep, sourceInfo, p); err != nil {
+		return errors.Wrapf(err, "can't save policy for %v", sourceInfo)
+	}
+
+	return nil
+}
+
+// ValidateSchedulingPolicy returns an error if manual field is set along with scheduling fields.
+func ValidateSchedulingPolicy(p SchedulingPolicy) error {
+	if p.IntervalSeconds != 0 && p.Manual {
```

Comment:
idea: this maybe more generic and future-proof:

if p.Manual && p != SchedulingPolicy{Manual: true} {
  return /* manual cannot be combined with other scheduling policies */
}